src/modules/avformat/producer_avformat.c

```diff
@@ -1019,7 +1041,8 @@ static int producer_open(producer_avformat self, mlt_profile profile, const char
 #ifdef AVFILTER
 				if (!test_open) {
 					self->autorotate = !mlt_properties_get(properties, "autorotate") || mlt_properties_get_int(properties, "autorotate");
-					error = setup_autorotate_filters(self);
+					self->filtergraph = mlt_properties_get(properties, "filtergraph");
```

Comment:
Why are you saving this value instead of simply getting the property when needed? For example, wherever you access `self->filtergraph` instead
```
const char *filtergraph = mlt_properties_get(properties, "filtergraph");
if ((self->autorotate  || self->filtergraph)...
```
below, or
`if (!error) error = (avfilter_graph_parse(self->vfilter_graph, filtergraph, inputs, outputs, NULL) < 0);`
above.

Reply:
I just wanted to mimic the behavior of dealing with autorotate but of course I can change it to a direct property access.

Reply:
I see. Upon reviewing `self->autorotate` I probably did it this because it was not a simple get property:
`self->autorotate = !mlt_properties_get(properties, "autorotate") || mlt_properties_get_int(properties, "autorotate");`
And then used from 4 different functions.